updated correct API to get index templates etc

ramdaspotale opened the pull request:
Hi, I think documentation needs correction on below points.
1. updated correct API to get index template
2.  on line 18, it should be index pattern instead of template pattern
3.  index pattern should have es version as 8 which matches elasticsearch version, also updated index patterns in API request payload

<!--
Thank you for your interest in and contributing to Elasticsearch! There
are a few simple things to check before submitting your pull request
that can help with the review process. You should delete these items
from your submission, but they are here to help bring them to your
attention.
-->

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

ramdaspotale committed:
updated correct API to get index templates etc

Hi, I think documentation needs correction on below points.
1. updated correct API to get index template
2.  on line 18, it should be index pattern instead of template pattern
3.  index pattern should have es version as 8 which matches elasticsearch version, also updated index patterns in API request payload

ramdaspotale commented:
pinging @elastic/es-docs

ramdaspotale commented:
@elasticsearchmachine

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

abdonpijpelink commented:
@ramdaspotale thank you for updating the docs! Your PR is targeting 8.6. To ensure this is also updated in future documentation, the PR should target the `main` branch instead. Can you rebase it, or would you like me to do it for you?

ramdaspotale commented:
@abdonpijpelink thanks for your update..i am okay if you can rebase it to main

abdonpijpelink commented on docs/reference/monitoring/indices.asciidoc:
```diff
@@ -28,7 +28,7 @@ and the number of replicas to two.
 ----------------------------------
 PUT /_template/custom_monitoring
```
We should also change this request:
- use the `_index_template` endpoint 
- `order` is now `priority` (need to change it in the text above, on line 19, too)
- the `settings` object needs to be wrapped in a `template` object

ramdaspotale replied:
I have updated the doc as per your suggestions. please verify

abdonpijpelink replied:
`settings` needs to be wrapped in a `template` object. The request becomes:

```
PUT /_index_template/custom_monitoring
{
  "index_patterns": [
    ".monitoring-beats-8-*",
    ".monitoring-es-8-*",
    ".monitoring-kibana-8-*",
    ".monitoring-logstash-8-*"
  ],
  "priority": 1,
  "template": {
    "settings": {
      "number_of_shards": 5,
      "number_of_replicas": 2
    }
  }
}
```

ramdaspotale replied:
please verify now

abdonpijpelink commented on docs/reference/monitoring/indices.asciidoc:
```diff
@@ -9,13 +9,13 @@ You can retrieve the templates through the `_template` API:
 monitoring indices. To override the default settings, add your own template:
 
-. Set the `template` patterns to match existing `.monitoring-{product}-7-*` indices.
+. Set the `index` patterns to match existing `.monitoring-{product}-8-*` indices.
```
```suggestion
. Set `index_patterns` to match existing `.monitoring-{product}-8-*` indices.
```

abdonpijpelink reviewed (changes requested):
This looks good. I've left some suggestions.

Also, on line 5, we should link to docs for the new index templates (change `<<indices-templates-v1` into `<<index-templates`)

ramdaspotale committed:
fix: grammatical correction to line 18

removed "the" from line 18 which is not very accuarate

Co-authored-by: Abdon Pijpelink <abdon.pijpelink@elastic.co>

ramdaspotale committed:
additional corrections to the docs as suggested by @abdonpijpelink

. use the `_index_template` endpoint instead of just `_template`
. change `order` to `priority`

ramdaspotale reviewed (commented):
changes done as requested.

ramdaspotale commented:
apologies..i missed that...i will add settings to template object.

ramdaspotale committed:
settings need to be wrapped inside template obect

ramdaspotale reviewed (commented):
`settings` added inside template object.

abdonpijpelink reviewed (approved):
LGTM. Thank you @ramdaspotale !

abdonpijpelink commented:
@elasticmachine test this please

ramdaspotale commented:
@abdonpijpelink  looks like builds finished

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [8.7](https://github.com/elastic/elasticsearch/pull/93940)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/93940">](https://github.com/elastic/elasticsearch/pull/93940) |
| ✅ |  [8.6](https://github.com/elastic/elasticsearch/pull/93941)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/93941">](https://github.com/elastic/elasticsearch/pull/93941) |
